executor/api/kafka/client.go

```diff
@@ -0,0 +1,152 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"github.com/seldonio/seldon-core/executor/api"
+	"github.com/seldonio/seldon-core/executor/api/client"
+	"github.com/seldonio/seldon-core/executor/api/payload"
+	"github.com/seldonio/seldon-core/executor/api/rest"
+	v1 "github.com/seldonio/seldon-core/operator/apis/machinelearning.seldon.io/v1"
+	"io"
+)
+
+type KafkaClient struct {
+	Hostname       string
+	DeploymentName string
+	Namespace      string
+	Protocol       string
+	Transport      string
+	predictor      *v1.PredictorSpec
+	Broker         string
+	Log            logr.Logger
+	topicHandlers  map[string]*KafkaRPC
+}
+
+func NewKafkaClient(hostname, deploymentName, namespace, protocol, transport string, predictor *v1.PredictorSpec, broker string, log logr.Logger) client.SeldonApiClient {
+	skc := &KafkaClient{
```

Comment:
Interesting, so this PR would also allow to enable kafka protocol for the internal graph data flow?

Reply:
It doesn't seem to be an option within the server.go, is this currently setup to be enabled for models?

Reply:
Its is enabled via kafka_full_graph option but not discussed in docs at present as is an advanced feature for which we need more validation and feedback.